lib/api/bucketPut.js

```diff
@@ -37,6 +37,14 @@ export default function bucketPut(authInfo, request, locationConstraint, log,
         log.trace('invalid acl header');
         return callback(errors.InvalidArgument);
     }
+    // - AWS JS SDK sends a request with locationConstraint us-east-1
+    // if no locationConstraint provided.
+    if (locationConstraint && config.locationConstraints &&
+      Object.keys(config.locationConstraints).indexOf(locationConstraint) < 0) {
+        log.trace('locationConstraint is invalid',
+          { locationConstraint });
+        return callback(errors.InvalidLocationConstraint);
+    }
```

Comment:
as discussed, if user does not send in a location, let's set it based on the default for the restEndpoint.